modules/fscloud/variables.tf

```diff
@@ -156,6 +156,21 @@ variable "existing_serviceref_zone" {
     {
       zone_id = string
   }))
+  validation {
+    condition = var.existing_serviceref_zone == null || (alltrue([for zone in var.existing_serviceref_zone : can(regex("^[0-9a-fA-F]{32}$", zone.zone_id))])
```

Comment:
1- There should be a seperate validation for both service ref and zone id.
2- This validation should be done on the map.